db/app.js

```diff
@@ -3,23 +3,31 @@ const {
   getTopics,
   getArticles,
   getArticlesById,
+  patchArticleById
 } = require("./controllers/controller");
 
 const app = express();
 app.use(express.json());
 app.get("/api/topics", getTopics);
 app.get("/api/articles/:article_id", getArticlesById);
 app.get("/api/articles", getArticles);
+app.patch("/api/articles/:article_id",patchArticleById);
 
 const customErr = app.use((err, req, res, next) => {
 if(err.status && err.msg) {
         res.status(err.status).send({ msg: err.msg })
 } else next(err);
 });
 
-const invalidID = app.use((err, req, res, next) => {
+const emptyPatch = app.use((err,req,res,next)=> {
+    if (err.code === '23502') {
+      res.status(400).send({ msg: "missing required fields" });
+    } else next(err);
+  });
+
+const stringInsteadOfNumber = app.use((err, req, res, next) => {
     if (err.code === '22P02') {
-      res.status(400).send({ msg: "Not a valid ID - this must be a number." });
+      res.status(400).send({ msg: "Incorrect type - this must be a number" });
     } else next(err);
   });
```

Comment:
I wouldn't assign the app.use error handlers to their own variables, feels like a bit more overengineering than needed

What I would consider is extracting the callbacks you pass into the app.use's into their own variables to keep things clearer - this might include extracting them into an `errors.js` file and importing them in here